RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelArrayBuffer.h

```diff
@@ -64,10 +68,17 @@ int SiPixelArrayBuffer::operator()(int row, int col) const { return pixel_vec[in
 int SiPixelArrayBuffer::operator()(const SiPixelCluster::PixelPos& pix) const { return pixel_vec[index(pix)]; }
 
 // unchecked!
-void SiPixelArrayBuffer::set_adc(int row, int col, int adc) { pixel_vec[index(row, col)] = adc; }
 
-void SiPixelArrayBuffer::set_adc(const SiPixelCluster::PixelPos& pix, int adc) { pixel_vec[index(pix)] = adc; }
+void SiPixelArrayBuffer::set_adc(int row, int col, int adc) { pixel_vec[index(row, col)] = std::min(adc, 65535); }
+
+void SiPixelArrayBuffer::set_adc(const SiPixelCluster::PixelPos& pix, int adc) {
+  pixel_vec[index(pix)] = std::min(adc, 65535);
+}
 
-void SiPixelArrayBuffer::add_adc(int row, int col, int adc) { pixel_vec[index(row, col)] += adc; }
+void SiPixelArrayBuffer::add_adc(int row, int col, int adc) {
+  auto const idx = index(row, col);
+  pixel_vec[idx] += adc;
+  pixel_vec[idx] = std::min(pixel_vec[idx], 65535);
```

Comment:
could you please use ```int(std::numeric_limits<uint16_t>::max())``` ?
btw how ```adc``` can be larger than that?  (I suspect a problem upstream)